### PR TITLE
GAX additions to support server streaming

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: src/Google.Api.Gax.Grpc/ApiServerStreamingCall.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        /// </summary>
+        /// <param name="request">The RPC request.</param>
+        /// <param name="perCallCallSettings">The call settings to apply to this specific call,
+        /// overriding defaults where necessary.</param>
+        /// <returns>A gRPC duplex streaming call object.</returns>
+        public AsyncServerStreamingCall<TResponse> Call(TRequest request, CallSettings perCallCallSettings)
+        {
+            CallSettings effectiveCallSettings = BaseCallSettings.MergedWith(perCallCallSettings);
+            GaxPreconditions.CheckState(effectiveCallSettings?.Timing?.Type != CallTimingType.Retry,
+                "Retry cannot be used in a server streaming method.");
```
Comment (jskeet) [inline src/Google.Api.Gax.Grpc/ApiServerStreamingCall.cs]: Are we sure about that? It makes more sense for a server-streaming call than a bidi call, but I don't know whether it actually works...

File: src/Google.Api.Gax.Grpc/ClientHelper.cs [hunk 2]
```diff
@@ -72,6 +72,25 @@ public ClientHelper(ServiceSettingsBase settings)
         }
 
         /// <summary>
+        /// Builds an <see cref="ApiServerStreamingCall"/> given a suitable underlying duplex call.
```
Comment (jskeet) [inline src/Google.Api.Gax.Grpc/ClientHelper.cs]: Nit: not duplex

File: test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs [hunk 3]
```diff
@@ -30,6 +30,21 @@ public void BuildApiCall_ClientSettings()
...
         [Fact]
+        public void BuildServerStreamingApiCall_ClientSettings()
+        {
+            var clientSettings = CallSettings.FromCancellationToken(new CancellationTokenSource().Token);
+            var helper = new ClientHelper(new DummySettings
+            {
+                CallSettings = clientSettings
+            });
+            var server = new DummyServerServerStreaming();
+            var apiCall = helper.BuildApiCall<SimpleRequest, SimpleResponse>(
```
Comment (jskeet) [inline test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs]: Unwrap line?

File: test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs [hunk 4]
```diff
@@ -81,6 +108,18 @@ public void BuildApiCall_PerCallSettings()
         }
 
         [Fact]
+        public void BuildServerStreamingApiCall_PerCallSettings()
+        {
+            var perCallSettings = CallSettings.FromCancellationToken(new CancellationTokenSource().Token);
+            var helper = new ClientHelper(new DummySettings());
+            var server = new DummyServerServerStreaming();
+            var apiCall = helper.BuildApiCall<SimpleRequest, SimpleResponse>(
```
Comment (jskeet) [inline test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs]: Unwrap line?

File: src/Google.Api.Gax.Grpc/ApiServerStreamingCall.cs (hunk 1 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax.Grpc/ApiServerStreamingCall.cs]: I'm fairly unsure about this. I've put in a todo to check this further.

File: src/Google.Api.Gax.Grpc/ClientHelper.cs (hunk 2 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax.Grpc/ClientHelper.cs]: Done

File: test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs (hunk 3 above)
Comment (chrisdunelm) [inline test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs]: Done

File: test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs (hunk 4 above)
Comment (chrisdunelm) [inline test/Google.Api.Gax.Grpc.Tests/ClientHelperTest.cs]: Done

**Pull request merged**